src/com/northvine/reader/implementation/DataMatrix.java

```
@@ -199,30 +202,22 @@ private void shiftImageLeft(int offset)
     * 
     * 
     * @return y coordinate of horizontal closed limitation line in
-    *         imageData array
+    *         imageData array, 0 if no true values in array
     */
    private int verticalScan()
    {
-      boolean blackChar = false;
-      int bottomIndex = 0;
       // Find the bottom
       for(int col = 0; col < barcodeImage.MAX_WIDTH; col++)
       {
          for(int row = barcodeImage.MAX_HEIGHT; row >= 0; row--)
          {
             if(barcodeImage.getPixel(row, col))
             {
-               bottomIndex = row;
-               blackChar = true;
-               break;
+               return row;
             }
          }
-         if(blackChar)
-         {
-            break;
-         }
       }
-      return bottomIndex;
+      return 0;

```

should we throw an exception, like IllegalStateException? I am wondering if this should ever or could ever happen?